docs/source/common/checkpointing_expert.rst

```diff
@@ -87,3 +87,10 @@ Custom Checkpoint IO Plugin
 .. note::
 
     Some ``TrainingTypePlugins`` like ``DeepSpeedStrategy`` do not support custom ``CheckpointIO`` as checkpointing logic is not modifiable.
+
+
+*********************************
+Writing your own Checkpoint class
+*********************************
```

Comment:
I would move this above the checkpoint io plugin. I predict that it is more likely or more intuitive that a user would want to override the callback first before looking at the checkpoin io api.

Reply:
Makes sense, moved